prov/verbs/src/fi_verbs.h

```diff
@@ -388,16 +388,21 @@ struct fi_ibv_cq {
 	struct util_buf_pool	*wce_pool;
 };
 
+struct fi_ibv_rdm_request;
+typedef void (*fi_ibv_rdm_cq_read_entry)(struct fi_ibv_rdm_request *cq_entry,
+					 int index, void *buf);
+
 struct fi_ibv_rdm_cq {
-	struct fid_cq		cq_fid;
-	struct fi_ibv_domain	*domain;
-	struct fi_ibv_rdm_ep	*ep;
-	struct dlist_entry	request_cq;
-	struct dlist_entry	request_errcq;
-	uint64_t		flags;
-	size_t			entry_size;
-	int			read_bunch_size;
-	enum fi_cq_wait_cond	wait_cond;
+	struct fid_cq			cq_fid;
+	struct fi_ibv_domain		*domain;
+	struct fi_ibv_rdm_ep		*ep;
+	struct dlist_entry		request_cq;
+	struct dlist_entry		request_errcq;
+	uint64_t			flags;
+	size_t				entry_size;
+	fi_ibv_rdm_cq_read_entry	read_entry;
+	int				read_bunch_size;
```

Comment:
Should this be size_t? I'm guessing it is defined as an int because ibv_poll_cq() expects an int, not size_t.

Reply:
Actually, yes, It should be `size_t`. Some places of code assumes that this var should `size_t` (e.x. `fi_cq_read`'s `count` parameter)
But `fi_param_get` set of functions that read value from the runtime parameters (env) doesn't support `size_t` at all.
I guess we will extend the `fi_param_get` to support more types. What do you think about this? But it should be a separate PR